Sources/MongoSwift/MongoClient.swift

```diff
@@ -252,41 +274,109 @@ public class MongoClient {
         self.decoder = BSONDecoder(options: options)
         self.sdamEventHandlers = []
         self.commandEventHandlers = []
+        self.state = .upAndRunning
+
         self.connectionPool.initializeMonitoring(client: self)
     }
 
     deinit {
-        assert(self.isClosed, "MongoClient was not closed before deinitialization")
+        assert(
+            self.state == .shutDown,
```

Comment:
since `deinit` is called when there are 0 references left to the client, I don't believe it's possible for there to be any races around access to this